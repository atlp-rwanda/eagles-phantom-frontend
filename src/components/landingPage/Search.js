import React, { Component } from 'react'
import '../../scss/search.scss';

export default class Search extends Component {
    render() {
        return (
    <form  className="search">
    <select className="origin"  value="Origin" onChange={()=>{value=this.value}}>
    <option disabled>Origin</option>
    <option >Nyabugogo</option>
    <option >Downtown</option>
    <option >Kacyiru</option>
    <option >Remera</option>
    </select>
    <select className="destination"  value="Destination" onChange={()=>{value=this.value}}>
    <option  disabled>Destination</option>
    <option >Kimironko</option>
    <option >Downtown</option>
    <option >Kacyiru</option>
    <option >Remera</option>
    </select>
    <button className="searchBtn" type="submit" ><i className="fa fa-search"></i> Search</button>
    </form>
        )
    }
}