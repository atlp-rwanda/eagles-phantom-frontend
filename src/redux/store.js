import {createStore, applyMiddleware } from 'redux';
import { composeWithDevTools} from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import allReducers from './reducers/';

// const rootReducer = combineReducers({
//   count: counterReducer,
// });

// const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const middleware = applyMiddleware(thunk, promise)
const store = createStore(allReducers, {}, composeWithDevTools(middleware));

export default store;
