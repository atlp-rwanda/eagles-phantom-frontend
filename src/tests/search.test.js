import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from "react-redux";
import store from "../redux/store.js";
import Search from '../components/landingPage/Search';

const search=()=>{
    return render(
       <Provider store={store}>
       <Search />
       </Provider>
    )
}

describe("Search",()=>{
    it("should render the component",()=>{
        const { asFragment }=search();
        expect(asFragment(<Search />)).toMatchSnapshot()
    });
})