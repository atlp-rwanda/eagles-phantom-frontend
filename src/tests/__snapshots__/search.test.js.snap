// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render the component 1`] = `
<DocumentFragment>
  <form
    class="search"
  >
    <select
      class="origin"
    >
      <option
        disabled=""
      >
        Origin
      </option>
      <option>
        Nyabugogo
      </option>
      <option>
        Downtown
      </option>
      <option>
        Kacyiru
      </option>
      <option>
        Remera
      </option>
    </select>
    <select
      class="destination"
    >
      <option
        disabled=""
      >
        Destination
      </option>
      <option>
        Kimironko
      </option>
      <option>
        Downtown
      </option>
      <option>
        Kacyiru
      </option>
      <option>
        Remera
      </option>
    </select>
    <button
      class="searchBtn"
      type="submit"
    >
      <i
        class="fa fa-search"
      />
       Search
    </button>
  </form>
</DocumentFragment>
`;
